src/main/java/org/truffleruby/core/rope/RopeNodes.java

```diff
@@ -46,9 +46,10 @@
 
 public abstract class RopeNodes {
 
+    // Preserves encoding of the top-level Rope
     @NodeChildren({
             @NodeChild(type = RubyNode.class, value = "base"),
-            @NodeChild(type = RubyNode.class, value = "offset"),
+            @NodeChild(type = RubyNode.class, value = "offset"), // in bytes
```

Comment:
My impression was that `offset` is always a byte offset and `charOffset` a character offset, but maybe that's untrue?
This is also basically a property of the node, either it takes byteOffset + byteLength or characterOffset + characterLength, but not a mix.